### PR TITLE
use custom SSO value for authTokenType -> to detect if account is added by SSO

### PR DESCRIPTION
**Pull request by tobiasKaminsky**
Fix: https://github.com/nextcloud/android/issues/7116
SSO: https://github.com/nextcloud/Android-SingleSignOn/pull/270

Signed-off-by: tobiasKaminsky <tobias@kaminsky.me>

### Testing 
Writing tests is very important. Please try to write some tests for your PR. 
If you need help, please do not hesitate to ask in this PR for help.

[unit tests](https://github.com/nextcloud/android/blob/master/CONTRIBUTING.md#unit-tests)
[instrumented tests](https://github.com/nextcloud/android/blob/master/CONTRIBUTING.md#instrumented-tests)
[UI tests](https://github.com/nextcloud/android/blob/master/CONTRIBUTING.md#ui-tests)

- [x] Tests written, or not not needed

### CODE REVIEW

**Comment by David-Development** [comment]
@tobiasKaminsky Doesn't this break backwards compatibility as older SSO libraries don't report that value yet?

**Comment by tobiasKaminsky** [comment]
> @tobiasKaminsky Doesn't this break backwards compatibility as older SSO libraries don't report that value yet?

It is used as "onlyAdd" value, which if is true finish AuthenticatorActivity after adding an new account.
As previously it was checked wrongly, it did something strange on Files app, but worked on 3rd party libs.
So yes, it will break it, but which only means that 3rd party libs without a new SSO release have to manually close Files app after adding a new account.

tl;dr: it is safe to merge

**Review by AndyScherzinger — APPROVED** [review]
